stages/test/test_kickstart.py

```diff
@@ -30,3 +31,41 @@ def test_kickstart(tmp_path, test_input, expected):
     with open(os.path.join(tmp_path, ks_path), encoding="utf-8") as fp:
         ks_content = fp.read()
     assert ks_content == expected + "\n"
+
+
+@pytest.mark.parametrize("test_data,expected_err", [
+    # BAD pattern, ensure some obvious ways to write arbitrary
+    # kickstart files will not work
+    ({"clearpart": {"disklabel": r"\n%pre\necho p0wnd"}}, r"p0wnd' does not match"),
+    ({"clearpart": {"drives": [" --spaces-dashes-not-allowed"]}}, "' --spaces-dashes-not-allowed' does not match"),
+    ({"clearpart": {"drives": ["\n%pre not allowed"]}}, "not allowed' does not match"),
+    ({"clearpart": {"list": ["\n%pre not allowed"]}}, "not allowed' does not match"),
+    ({"clearpart": {"disklabel": "\n%pre not allowed"}}, "not allowed' does not match"),
+    # GOOD pattern we want to keep working
+    ({"clearpart": {"drives": ["sd*|hd*|vda", "/dev/vdc"]}}, ""),
+    ({"clearpart": {"drives": ["disk/by-id/scsi-58095BEC5510947BE8C0360F604351918"]}}, ""),
+    ({"clearpart": {"list": ["sda2","sda3","sdb1"]}}, ""),
+])
+def test_schema_validation_smoke(test_data, expected_err):
+    name = "org.osbuild.kickstart"
+    root = os.path.join(os.path.dirname(__file__), "../..")
```

Comment:
This could be extracted into a osbuild.testutil helper like "get_schema_for_stage" or something but given how small it is that feels like overkill.